src/mpi/request/waitall.c

```diff
@@ -80,11 +80,14 @@ int MPIR_Waitall(int count, MPI_Request array_of_requests[], MPI_Status array_of
     int n_completed;
     int active_flag;
     int rc = MPI_SUCCESS;
-    int n_greqs;
     int proc_failure = FALSE;
     int disabled_anysource = FALSE;
     const int ignoring_statuses = (array_of_statuses == MPI_STATUSES_IGNORE);
-    int optimize = ignoring_statuses;   /* see NOTE-O1 */
+    /* Optimize meaning:
+     *  0> generic request/status and there is generalized requests
+     *  1> generic request/status and there is no generalized requests
+     *  2> non-null and Send/Recv request and ignoring status */
+    int optimize = ignoring_statuses ? 2 : 1;
```

Comment:
Shouldn't we define macro constants instead of 0,1,2...?

Reply:
Yes I am redoing these waitall patches